langchain/tools/base.py

```diff
@@ -24,6 +24,8 @@ class BaseTool(ABC, BaseModel):
     name: str
     description: str
     args_schema: Optional[Type[BaseModel]] = None
+    """An optional filter function that determines whether the input to the tool is appropriate."""
```

Comment:
@hwchase17 We've been back and forth on this a bit, but this could technically be added as validators in the args_schema now that that's added. 

Pros of the proposed approach is it's more discoverable. 
Cons is that we have two layers of validation. and add more arguments to the tool class.